samples: subsys: usb: mass: Add stm32l562e_dk_ns support

nandojve opened the pull request:
Add stm32l562e_dk_ns board support. It adds support to build images with TFM. The documentation was update to mention the stm32l562e_dk board build steps.

Tested with 2GB transflash card using FAT32

depends on #49983 and #49984

erwango reviewed (changes requested):
Thanks for sharing this work.
Can you also put the board usb support. If I'm correct this is not available yet.

nandojve commented:
> Thanks for sharing this work. Can you also put the board usb support. If I'm correct this is not available yet.

Hi @erwango ,

Could you revisit and check if everything is fine now?

BTW, any clue about error from tests/arch/arm/arm_thread_swap_tz?

Easy to reproduce with `west build -b stm32l562e_dk_ns tests/arch/arm/arm_thread_swap_tz`

erwango commented on drivers/disk/Kconfig.sdmmc:
```diff
@@ -47,7 +47,10 @@ config SDMMC_STM32
 	bool "STM32 SDMMC Hardware Flow control"
 	depends on SDMMC_STM32
-	depends on SOC_SERIES_STM32H7X || SOC_SERIES_STM32F7X || SOC_SERIES_STM32L4X
+	depends on SOC_SERIES_STM32H7X || \
+		   SOC_SERIES_STM32F7X || \
+		   SOC_SERIES_STM32L4X || \
+		   SOC_SERIES_STM32L5X
```
Can you replace by a check on "st,stm32-sdmmc" using dt_compat_enabled? This will avoid adding the same modification each time a new series is added to this driver.

erwango replied:
ok, just seeing that this is `HWFC` no the driver symbol. Please discard this comment.

erwango commented on samples/subsys/usb/mass/boards/stm32l562e_dk_ns.overlay:
```diff
@@ -0,0 +1,51 @@
+	};
+};
+
+&sdmmc1 {
+	status = "okay";
+
+	pinctrl-0 = <&sdmmc1_d0_pc8
+		     &sdmmc1_d1_pc9
+		     &sdmmc1_d2_pc10
+		     &sdmmc1_d3_pc11
+		     &sdmmc1_ck_pc12
+		     &sdmmc1_cmd_pd2>;
+
+	pinctrl-names = "default";
+
+	cd-gpios = <&gpiof 2 GPIO_ACTIVE_LOW>;
+};
```
Move this to board description (ns/s common part)

nandojve replied:
Done!

erwango commented on samples/subsys/usb/mass/boards/stm32l562e_dk_ns.overlay:
```diff
@@ -0,0 +1,51 @@
+
+&flash0 {
+
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 0x00019000>;
+			read-only;
+		};
+		/* Secure image primary slot */
+		slot0_partition: partition@19000 {
+			label = "image-0";
+			reg = <0x00019000 0x0003C000>;
+		};
+		/* Non-secure image primary slot */
+		slot1_partition: partition@55000 {
+			label = "image-1";
+			reg = <0x00055000 0x0002A000>;
+		};
```
Can this be moved to board description ?

nandojve replied:
Done!

erwango commented:
> Hi @erwango ,
> 
> Could you revisit and check if everything is fine now?

cf few comments. Can you convert this to a ready to review PR?

> BTW, any clue about error from tests/arch/arm/arm_thread_swap_tz?
> 
> Easy to reproduce with `west build -b stm32l562e_dk_ns tests/arch/arm/arm_thread_swap_tz`

We've seen this recently. Can you open a dedicated issue ?

nandojve commented:
CI is failing because of lpcxpresso55s69 and it seems not related to this PR.

stephanosio commented:
> CI is failing because of lpcxpresso55s69 and it seems not related to this PR.

@mmahadevan108 ^^

erwango commented on dts/arm/st/l5/stm32l5.dtsi:
```diff
@@ -337,6 +337,15 @@
+			clocks = <&rcc STM32_CLOCK_BUS_AHB2 0x00400000>;
+			interrupts = <78 0>;
+			status = "disabled";
+			label = "SDMMC_1";
```
Labels should now be removed.
cf https://github.com/zephyrproject-rtos/zephyr/pull/48358

jfischer-no commented on samples/subsys/usb/mass/README.rst:
```diff
@@ -122,6 +122,26 @@ follows:
 
+For boards with TFM support a Non-Secure image can be build:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/usb/mass
+   :board: stm32l562e_dk_ns
+   :goals: build
+   :compact:
+
+.. note::
+   Note that ``arm-none-eabi-gcc`` should be available in the PATH variable
+   and that ``STM32_Programmer_CLI`` is required to run ``regression.sh``
+   (see https://www.st.com/en/development-tools/stm32cubeprog.html). If you
+   are still having trouble running these scripts, check the Programming and
+   Debugging section of the :ref:`stm32l562e_dk_board` documentation.
+
+.. code-block:: console
+
+   ./build/tfm/regression.sh
+   west flash
```
Honestly, I want to keep all samples generic, that is not quite the case for this one, but this changes going too far in my opinion, too board and configuration specific, sorry.

erwango replied:
I think this should just be cleaned up as this is not sample specific and is already documented at board level.

nandojve commented:
Add DNM until #49983 and #49984 got merged.

nandojve commented:
Rebase after #49983 and #49984 got in and drop DNM flag.

nandojve commented:
rebase due to conflict

nandojve commented:
Rebase after #51852.

nandojve committed:
boards: arm: stm32l562e_dk: Enable usb tags for tests

This enables usb and usb_device tag for tests. It explicitly disable
samples/subsys/usb/dfu for non-secure version of this board due to
building errors, which can be explored in future.

Signed-off-by: Gerson Fernando Budke <gerson.budke@ossystems.com.br>

nandojve committed:
samples: usb: mass: Add support to TFM NS boards

This add TFM cmake definitions to build mass storage sample for non
secure firmware boards versions. The stm32l562e_dk_ns board was used
to demonstrate. It uses SDMMC/SD with TFM profile large. Tests were
conducted using a 2GB transflash card with FAT-32 partition.

Signed-off-by: Gerson Fernando Budke <gerson.budke@ossystems.com.br>
